portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,83 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import java.util.Arrays;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private Comment userComment;
+    private final ArrayList<String> comments = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+    
+    // Tutorial Steps Below:
+    // ArrayList<String> messages = new ArrayList<String>();
+    // messages.add("Alpha");
+    // messages.add("Beta");
+    // messages.add("Charlie");
+    
+    // String json = convertToJsonUsingGson(messages);
+
+    // response.setContentType("application/json");
+    // response.getWriter().println(json);
+
+    //response.setContentType("text/html;");
+    //response.getWriter().println("Hello Andrew");
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, "name-input", "");
+    String content = getParameter(request, "text-input", "");
+    boolean owo = Boolean.parseBoolean(getParameter(request, "owo", "false"));
```

Comment:
Theres a couple of related improvements I see here.

1. You might want to make a separate function for getting boolean parameters. What is happening now is you are creating the string "false" and using it as the default, but the only way which that value will be used is to be parsed by parseBoolean and be turned into a boolean. It would be more preferable to only ever use a boolean type itself than create a string that is intended to parse into a boolean. For exmaple, if I am using your code, and follow the same pattern I might accidentally make a typo and enter "fasle" and the problem is the code would still compile and I would never know. I would only know when a user of my service hits the endpoint and it fails, which leads me to my next improvement:
1. What happens if owo is set but its set to "hello" instead of either "true" or "false"? In what ways does this fail if the user provides bad input? How can you be more informative about what went wrong when it does fail?
hint: using a try-catch to catch the exception and either a. rethrow a more informative exception or b. return a 400 level error [more complicated but more correct]

Reply:
Took a bit but I first used a try and block in my helper methods. I realized that it's more efficient to put it in my doPost method. I was able to handle bad inputs by showing the 400 page error if the user puts a string of words instead of a number.